oaknational/Oak-Web-Application
# Feat/lesq 729/legacy search [LESQ-729]

## Description

Music year: 1953

- remove call to 2020 search api
- update legacy filter to exclude EYFS only
- remove all 2020 search api code
- update all tests to use 2023 fixture and refactor some functionality to be shared

## How to test

Once the search API changes have been deployed to production:

1. Go to https://deploy-preview-2357--oak-web-application.netlify.thenational.academy
2. Search for eg 'volcanoes'
3. You should see legacy results, once each (no duplicates)
4. in the requests tab you should only see requests to one search API
5. you should still see results for non-legacy content
6. if you search for eg 'veggies' you should get no results (EYFS content)

### Screenshots:
the results are coming back in a different order for legacy content, which suggests they're being weighted differently. Not sure we care at this point as it brings them in line with new content? Compare with prod vs this branch

Eg using legacy api:
<img width="600" alt="Screenshot 2024-04-11 at 10 03 31" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/abb50cd4-1b0f-43d0-9409-ac24f1f753c1">


vs using new api:
<img width="600" alt="Screenshot 2024-04-11 at 10 05 02" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/e62420c8-ce6e-4454-aeea-a2d6a8ee9d26">
